### ISSUE TITLE
UP032: detect more cases

### ISSUE THREAD

**Issue by spaceone** [astral-sh/ruff#2031]
UP032 could detect more cases when `.format()` is used and suggest to use a fstring.

```
'Magic wand: {}'.format(bag['wand'])
'Magic wand: {}'.format(bag["wand"])
'Magic wand: {}'.format('bag')
'Magic wand: {}'.format(bag)
```
(only the last is detected).

A fixer for all cases is of course not trivial: https://peps.python.org/pep-0701/

**Comment by charliermarsh** [comment]
IIRC we only flag what we can confidently fix. We could instead flag _all_ usages of `.format`, even those we can't fix right now...

**Comment by Avasam** [comment]
I found a special case today that I think should be safe to fix:
`"  reading {filename}".format(**locals())`
I don't see why fixing to `f"  reading {filename}"` shouldn't be safe as `filename` should necessarily be in the local scope (otherwise the code would already fail anyway)

**Comment by Avasam** [comment]
Here's another one I found today in setuptools again:
```py
    '%(foo)s:%(bar)s' % vars(x)
    # can be
    f'{x.foo}:{x.bar}'

    # original example I found:
    def __str__(self):
        return '%(username)s:%(password)s' % vars(self)
```

**Comment by tdulcet** [comment]
UP031 does not flag this, although I thought it was supposed to detect everything after #11229:
```py
lines.extend(('┬', f'│ {l}', '├─%s─' % (len(l) * '─'), '│'))
```
After manually updating it use `str.format`, it is then not detected by UP032:
```py
lines.extend(('┬', f'│ {l}', '├─{}─'.format(len(l) * '─'), '│'))
```
The issue seems to be the string multiplication, as removing that causes it to be detected by both rules respectively as expected. This of course should have no effect on UP031, as it would not need to embed the strings and thus change the quote style like with UP032.

**Comment by inoa-jboliveira** [comment]
> IIRC we only flag what we can confidently fix. We could instead flag _all_ usages of `.format`, even those we can't fix right now...

Yes please. I don't believe there is a single format op that cannot be expressed as f-string. Specially with python 3.12 changes regarding quotes and backslashes.

So at least give us the warnings even if no fix exists. That is the purpose of a checker. IMO not showing a warn for `"{}".format("a")` should be considered a bug.
